istioctl/cmd/proxystatus.go

```diff
@@ -71,6 +87,15 @@ Retrieves last sent and last acknowledged xDS sync from Istiod to each Envoy in
 			}
 			if len(args) > 0 {
 				podName, ns := handlers.InferPodInfo(args[0], handlers.HandleNamespace(namespace, defaultNamespace))
+
+				path := fmt.Sprintf("/debug/config_dump?proxyID=%s.%s", podName, ns)
+				istiodDumps, err := kubeClient.AllDiscoveryDo(context.TODO(), istioNamespace, path)
+				if err != nil {
+					return err
+				}
+
+				// Contacting Envoy /config_dump can trigger it to disconnect from Istiod.  Don't
```

Comment:
?!

Reply:
I was surprised too.  My stale pod was returning a 404 when I investigated further.  I thought it was an Istiod bug because I could consistently use `ps` to list and see 9 pods, but when I asked for details it only looked at 8 of them.  I spent an hour looking at XDS internals before I realized what was happening.

It seems that if Envoy has rejected LDS, and it is asked to config_dump, it closes the connection and tries again.  That brief closure happens at the same time that _istioctl_ asks Pilot for `/debug/config_dump`.

Reply:
This sounds like we should raise an issue with Envoy?

Reply:
@liamawhite Read for re-review.

I suspect Envoy reconnecting with Istiod only happens when `ps` says the pod is in `STALE (Never Acknowledged)` state.  If I am characterizing it correctly, there isn't anything wrong with Envoy hanging up on XDS and trying again when asked to present its own `/config_dump`.  It surprised me but I could imagine it being the intended behavior.

Reply:
Am I missing something? It seems like a flaw that you can force envoy to disconnect from `istiod` by hitting `/config_dump`. Isn't that a potential DoS vector?

Reply:
@liamawhite Fair enough.  I will raise an issue with a working group to discuss.  Usually I go to Networking.  Do you suggest Networking or Security?

Because Pilot never rejects `istioctl proxy-status` for SYNCed and not always for STALE I am tempted I don't think we could DoS a working Envoy.  But I will follow up.

Reply:
Networking probably makes sense. I agree it's probably difficult to actually weaponize it but it definitely seems like a bug.

Reply:
I could not reproduced. Applied this diff to easily get a NACK:

```diff
diff --git a/pilot/pkg/networking/core/v1alpha3/listener.go b/pilot/pkg/networking/core/v1alpha3/listener.go
index 359e229941..1daac49bb9 100644
--- a/pilot/pkg/networking/core/v1alpha3/listener.go
+++ b/pilot/pkg/networking/core/v1alpha3/listener.go
@@ -2182,7 +2182,7 @@ func buildListener(opts buildListenerOpts) *listener.Listener {
 // TODO(lambdai): Calculate the filter chain match to replace the wildcard and replace appendListenerFallthroughRoute.
 func (configgen *ConfigGeneratorImpl) appendListenerFallthroughRouteForCompleteListener(l *listener.Listener, node *model.Proxy, push *model.PushContext) {
        for _, fc := range l.FilterChains {
-               if isMatchAllFilterChain(fc) {
+               if isMatchAllFilterChain(fc) && false {
                        // We can only have one wildcard match. If the filter chain already has one, skip it
                        // This happens in the case of HTTP, which will get a fallthrough route added later,
                        // or TCP, which is not supported
```

Then hit /config_dump and nothing happened

Reply:
@howardjohn Reproduce against httpbin with this:

```
kubectl apply -f - <<EOF
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: httpbin
spec:
  configPatches:
  - applyTo: HTTP_FILTER
    match:
      context: SIDECAR_INBOUND
      listener:
        filterChain:
          filter:
            name: envoy.http_connection_manager
            subFilter:
              name: envoy.router
    patch:
      operation: INSERT_BEFORE
      value:
        name: myfilter
        typed_config:
          '@type': type.googleapis.com/udpa.type.v1.TypedStruct
          type_url: type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
          value:
            config:
              configuration: world
              root_id: add_header
              vm_config:
                code:
                  local:
                    inline_bytes: !!binary |
                      AUkMYAJ/fwE=
                runtime: envoy.wasm.runtime.v8
                vm_id: myfilter
  workloadSelector:
    labels:
      app: httpbin
EOF
```

Reply:
I can reproduce

Reply:
File an issue in envoy? Config dump is very complicated internally.

Reply:
https://github.com/istio/istio/issues/26119